tools/webpack/interactivity.js

```diff
@@ -12,18 +12,28 @@ module.exports = {
 	...baseConfig,
 	name: 'interactivity',
 	entry: {
-		index: {
-			import: `./packages/interactivity/src/index.js`,
-			library: {
-				name: [ 'wp', 'interactivity' ],
-				type: 'window',
-			},
-		},
+		index: `./packages/interactivity/src/index.js`,
+		navigation: './packages/block-library/src/navigation/view.js',
+		query: './packages/block-library/src/query/view.js',
+		image: './packages/block-library/src/image/view.js',
+		file: './packages/block-library/src/file/view.js',
+		search: './packages/block-library/src/search/view.js',
+	},
+	experiments: {
+		outputModule: true,
 	},
 	output: {
 		devtoolNamespace: 'wp',
 		filename: './build/interactivity/[name].min.js',
+		library: {
+			type: 'module',
+		},
 		path: join( __dirname, '..', '..' ),
+		environment: { module: true },
+	},
+	externalsType: 'module',
+	externals: {
+		'@wordpress/interactivity': '@wordpress/interactivity',
```

Comment:
I think at some point in our config we need to maintain an esmodules array to keep these as is in the code (just like we have a dedicated config for bundled modules). I expect interactivity to be just one of these modules

Reply:
Yeah. I didn't pay much attention to the bundling side yet as it shouldn't affect the PHP API, so I just made it work 😄

_(I'm not a fan of the code produced by Webpack for ES modules, by the way.)_